.github/workflows/ci-main.yml

```
@@ -702,11 +699,12 @@ jobs:
               - "**/*.lock"
       - name: Run Cargo Clippy Linux
         id: cargo-clippy-linux
+        env:
+          BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
         if: |
           steps.filter.outputs.cargo == 'true' ||
-          github.event_name == 'schedule' ||
-          github.event_name == 'workflow_dispatch' ||
-          github.event_name == 'workflow_call'
+          env.BRANCH_NAME == 'master' ||
+          startsWith(env.BRANCH_NAME, 'rc--')
```

What about including `master-private` and `hotfix-` as well like:
```suggestion
          env.BRANCH_NAME == 'master' ||
          env.BRANCH_NAME == 'master-private' ||
          startsWith(env.BRANCH_NAME, 'rc--') ||
          startsWith(env.BRANCH_NAME, 'hotfix-')
```